java/maven/src/org/netbeans/modules/maven/output/DefaultOutputProcessorFactory.java

```diff
@@ -33,13 +33,13 @@
 public class DefaultOutputProcessorFactory implements ContextOutputProcessorFactory {
     
     @Override public Set<OutputProcessor> createProcessorsSet(Project project) {
-        Set<OutputProcessor> toReturn = new HashSet<OutputProcessor>();
+        Set<OutputProcessor> toReturn = new HashSet<>();
         if (project != null) {
             toReturn.add(new JavadocOutputProcessor());
             toReturn.add(new TestOutputListenerProvider());
             toReturn.add(new SiteOutputProcessor(project));
             NbMavenProjectImpl nbprj = project.getLookup().lookup(NbMavenProjectImpl.class);
-            toReturn.add(new ExecPluginOutputListenerProvider(nbprj));
```

Comment:
At least deprecate the original?

Was about to suggest making one extend the other, to reduce duplication, though the impl is pretty small anyway.

Reply:
sure I can do that if you want me to

I didn't deprecate because:
 - the class still works perfectly fine
 - I usually don't ever deprecate anything unless it can be deprecated for removal and NB doesn't really have a process for removing code (unfortunately), although it happened before e.g. with Schlieman or CND cluster etc

To reuse the code of the method I could put it into an abstract class and let both extend it. The only difference between both would be the goal array.

What would you prefer?

Reply:
i deprecated ExecPluginOutputListenerProvider and made the new JavaStacktraceOutputProcessor configurable. It uses global scope by default. Ran a project wide search and there were no usages of the deprecated class.

changed the hierarchy to make it DRY